modules/ingestor/src/service/cve/cve_record/v5.rs

```diff
@@ -24,6 +24,34 @@ impl CveMetadata {
             CveMetadata::Rejected(inner) => &inner.cve_id,
         }
     }
+
+    pub fn date_reserved(&self) -> Option<OffsetDateTime> {
+        match self {
+            CveMetadata::Published(inner) => inner.date_reserved,
+            CveMetadata::Rejected(inner) => inner.date_reserved,
+        }
+    }
+
+    pub fn date_published(&self) -> Option<OffsetDateTime> {
+        match self {
+            CveMetadata::Published(inner) => inner.date_published,
+            CveMetadata::Rejected(inner) => inner.date_published,
+        }
+    }
+
+    pub fn date_updated(&self) -> Option<OffsetDateTime> {
+        match self {
+            CveMetadata::Published(inner) => inner.date_updated,
+            CveMetadata::Rejected(inner) => inner.date_updated,
+        }
+    }
+
+    pub fn date_rejected(&self) -> Option<OffsetDateTime> {
+        match self {
+            CveMetadata::Published(_) => None,
+            CveMetadata::Rejected(inner) => inner.date_rejected,
+        }
+    }
```

Comment:
I still don't understand why the `CveMetadata` enum exists.

Reply:
Yeah, it would be clearer (to me) if the common fields were factored out of each.